src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -363,15 +363,17 @@ export class ConnectionConfig {
 	 */
 	public canChangeConnectionConfig(profile: ConnectionProfile, newGroupID: string): boolean {
 		let profiles = this.getIConnectionProfileStores(true);
-		let existingProfile = profiles.find(p =>
-			p.providerName === profile.providerName &&
-			this.checkIfAuthenticationOptionsMatch(p, profile) &&
-			p.options.databaseName === profile.options.databaseName &&
-			p.options.serverName === profile.options.serverName &&
-			p.options.userName === profile.options.userName &&
-			p.options.connectionName === profile.options.connectionName &&
-			p.groupId === newGroupID &&
-			this.checkIfNonDefaultOptionsMatch(p, profile));
+		let existingProfile = profiles.find(p => {
```

Comment:
Should we add tests for this?

Reply:
I think there's already tests for the comparison feature, I need to address the existing ones first as they're failing

Reply:
This is already part of the connection config tests